### PR TITLE
Set CMake python from sys.executable in setup.py

### PR DESCRIPTION
**Pull request by tgaddair**
cc @richardliaw

### CODE REVIEW

**Review by nvcastet — COMMENTED** [review]
if PYTHON_EXECUTABLE is not set, can you call
`find_package(Python 3.6 COMPONENTS Interpreter REQUIRED)`
Like that it still work outside of pip with newer version of cmake.

**Comment by tgaddair** [comment]
Thanks @nvcastet, let me know if the updated code is what you had in mind.

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,18 +97,10 @@ foreach(VAR in ITEMS HOROVOD_GPU_ALLREDUCE HOROVOD_GPU_ALLGATHER HOROVOD_GPU_BRO
...
-    execute_process(COMMAND bash -c "for p in $(echo $PATH | tr ':' ' '); do if find -E $p -maxdepth 1 -regex '.*/python3(\\.[0-9]*)?' -exec basename {} \; | sort -t. -k 2,2n | tail -1 | grep .; then break; fi done"
-                    OUTPUT_VARIABLE PY_EXE OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
-else()
-    execute_process(COMMAND bash -c "for p in $(echo $PATH | tr ':' ' '); do if find $p -maxdepth 1 -regex '.*/python3\\(\\.[0-9]*\\)?' -exec basename {} \; | sort -t. -k 2,2n | tail -1 | grep .; then break; fi done"
-                    OUTPUT_VARIABLE PY_EXE OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_QUIET)
-endif()
-if (NOT PY_EXE)
-     set(PY_EXE ${PYTHON_EXECUTABLE})
+if (NOT PYTHON_EXECUTABLE)
+    find_package(Python 3.6 COMPONENTS Interpreter REQUIRED)
```
[the remark this answers is not in the document]
Comment (tgaddair) [reply CMakeLists.txt:101]: Ah, case sensitive.  Okay, let me update that.

[the remark this answers is not in the document]
Comment (tgaddair) [reply CMakeLists.txt:101]: Okay, should be good now.

**Pull request merged**